samples/siwx91x_ota/src/main.c

```diff
@@ -0,0 +1,838 @@
+	if (ret) {
+		return ret;
+	}
+	if (k_event_wait_safe(&ctx->events, WLAN_EVENT_READY, false, K_MSEC(30000)) == 0) {
+		printf("WiFi scan failed or timed out\n");
+		return -errno;
+	}
+	return 0;
+}
+
+/*
+ * @brief Connect to WiFi network
+ *
+ * @param ssid Network SSID
+ * @param pwd Network password
+ * @param ctx App context pointer
+ * @return int 0 on success, negative error code on failure
+ */
+static int ota_wifi_connect(const char *ssid, const char *pwd, struct app_ctx *ctx)
+{
+	int ret;
+	struct net_if *iface = net_if_get_first_wifi();
+	struct wifi_connect_req_params cnx_params = {
+		.channel = WIFI_CHANNEL_ANY,
+		.band = 0,
+		.security = ctx->security_type,
+		.psk_length = strlen(pwd),
+		.psk = (uint8_t *)pwd,
+		.ssid_length = strlen(ssid),
+		.ssid = (uint8_t *)ssid
+	};
+
+	if (!ssid || !pwd) {
+		printf("Invalid connection parameters\n");
+		return -EINVAL;
+	}
+
+	printf("Connecting to network: %s\n", ssid);
+	ret = net_mgmt(NET_REQUEST_WIFI_CONNECT, iface, &cnx_params,
+		       sizeof(struct wifi_connect_req_params));
+	if (ret) {
+		printf("Connection request failed with error: %d\n", ret);
+		return ret;
+	}
+	if (k_event_wait_safe(&ctx->events, WLAN_EVENT_READY, false, K_MSEC(30000)) == 0) {
+		return -errno;
+	}
+	return 0;
+}
+
+/*
+ * @brief Wait for IP configuration to complete
+ * @param ctx Pointer to application context structure
+ * @return int 0 on success, negative error code on failure
+ */
+static int ota_wait_for_ip_config(struct app_ctx *ctx)
+{
+	int ret;
+
+	if (IS_ENABLED(CONFIG_NET_IPV4)) {
+		ret = k_event_wait_safe(&ctx->events, NET_IPV4_EVENT_READY, false, K_SECONDS(30));
+	} else {
+		ret = k_event_wait_safe(&ctx->events, NET_IPV6_EVENT_READY, false, K_SECONDS(30));
+	}
+	if (!ret && IS_ENABLED(CONFIG_NET_IPV6)) {
+		ret = k_event_wait_safe(&ctx->events, NET_IPV6_EVENT_READY, false, K_NO_WAIT);
+	}
+	return (ret > 0) ? 0 : -ETIMEDOUT;
+}
+
+/*
+ * @brief Connect to OTA server
+ *
+ * @param ctx Pointer to application context structure
+ * @return int 0 on success, negative error code on failure
+ */
+static int ota_connect_to_server(struct app_ctx *ctx)
+{
+	int ret;
+	struct zsock_addrinfo *res;
+	struct http_parse_data http_parse_data_st = ctx->http_parse_data_st;
+	int peer_verify = TLS_PEER_VERIFY_NONE;
+
+	ret = zsock_getaddrinfo(http_parse_data_st.host, http_parse_data_st.port, NULL, &res);
+	if (ret != 0) {
+		printf("Address resolution failed: %d\n", ret);
+		return ret;
+	}
+
+	if (http_parse_data_st.is_tls_enabled) {
+		ctx->sock = zsock_socket(res->ai_family, res->ai_socktype, IPPROTO_TLS_1_2);
+		if (ctx->sock < 0) {
+			goto error;
+		}
+		ret = zsock_setsockopt(ctx->sock, SOL_TLS, TLS_PEER_VERIFY, &peer_verify,
+				       sizeof(peer_verify));
+
+		if (ret < 0) {
+			goto error;
+		}
+
+		/* NOTE: The customer must not proceed to production with this */
+		ret = zsock_setsockopt(ctx->sock, SOL_TLS, TLS_HOSTNAME, NULL, 0);
```

Comment:
I believe this is redundant with `TLS_PEER_VERIFY_NONE`. You can drop this line.